openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java

```diff
@@ -344,6 +347,18 @@ else if (ALT_ROUTE.equalsIgnoreCase(algoStr))
                         ghRsp.addDebugInfo("tmode:" + tMode.toString());
                     }
                 }
+				//TODO make this nicer
+				try {
+					if (hints.has("user_speeds") && weighting instanceof AbstractSpeedCalculatorWeighting) {
+						AbstractSpeedCalculatorWeighting weighting1 = (AbstractSpeedCalculatorWeighting) weighting;
+						if (weighting1.getSpeedCalculator() instanceof RoadPropertySpeedCalculator) {
+							((RoadPropertySpeedCalculator) weighting1.getSpeedCalculator()).setRoadPropertySpeedMap((RoadPropertySpeedMap) ((ORSPMap) request.getAdditionalHints()).getObj("user_speeds"));
+						}
```

Comment:
i've also changed `weigthing1` to `weighting` here but without any difference.
Not sure which one is correct but i assume `weighting`. Probably was used to switch between the different weightings locally?